test/uri.jl

```diff
@@ -511,4 +511,8 @@ urltests = URLTest[
         checknp("http://a/b/c/../../", "http://a/")
         checknp("http://a/b/c/../../g", "http://a/g")
     end
+
+    @testset "joinpath" begin
+        @test joinpath(URIs.URI("http://a.b.c/d/e/f"), "a/b", "c") == URI("http://a.b.c/d/e/f/a/b/c")
```

Comment:
Would be good to have tests for any normalization we do with `_trailing_first_slash` etc.